discovery/discovery.go

```diff
@@ -140,5 +166,15 @@ func (o *orchestratorPool) GetOrchestrators(numOrchestrators int, suspender comm
 }
 
 func (o *orchestratorPool) Size() int {
-	return len(o.uris)
+	return len(o.infos)
+}
+
+func (o *orchestratorPool) SizeWith(scorePred common.ScorePred) int {
+	var size int
+	for _, info := range o.infos {
```

Comment:
There's an occassional panic on this line, that I suspect is due to `orchestratorPool == nil`